pkg/proxy/ipvs/proxier_test.go

```diff
@@ -3817,6 +3817,31 @@ func TestEndpointSliceE2E(t *testing.T) {
 	fp.servicesSynced = true
 	fp.endpointsSynced = true
 	fp.endpointSlicesSynced = true
+	expectedIPTablesWithSlice := `*nat
```

Comment:
I think it would be valuable to test this sort of behavior in a dedicated test function as opposed to adding to the EndpointSlice one. Based on the test comments above, `TestEndpointSliceE2E` was mainly meant for testing EndpointChangeTracker and EndpointSliceCache.

Reply:
I think a good starting point would be to copy a test like TestMasqueradeRule, but then modify it to check the entire iptables buffer, instead of just the masquerade rules.

Reply:
I see, will do that. Btw do you see anything missing for ipvs or ipset? If those are also missing then I can also group them in this changeset.

Reply:
Those should be covered in the other tests, it's mainly iptables rules we don't really test at the moment.

Reply:
Btw @andrewsykim  although I'm gonna change the test, it's weird to have multiple `:KUBE-POSTING - [0:0]` stat items in the output. I think it was hard coded [here](https://github.com/kubernetes/kubernetes/blob/v1.20.0-alpha.3/pkg/proxy/ipvs/proxier.go#L1881-L1893):
```golang
		if ch.table == utiliptables.TableNAT {
			if chain, ok := existingNATChains[ch.chain]; ok {
				writeBytesLine(proxier.natChains, chain)
			} else {
				writeLine(proxier.natChains, utiliptables.MakeChainLine(kubePostroutingChain))
			}
		} else {
			if chain, ok := existingFilterChains[KubeForwardChain]; ok {
				writeBytesLine(proxier.filterChains, chain)
			} else {
				writeLine(proxier.filterChains, utiliptables.MakeChainLine(KubeForwardChain))
			}
		}
```
Not sure if this is intended? If not it's easy to fix and I can do that.